tests/test_api_hosts_delete.py

```
@@ -238,6 +240,23 @@ def test_delete_hosts_chunk_size(
     query_wraper.query.limit.assert_called_with(5)
 
 
+def test_delete_stops_after_kafka_producer_error(
+    kafka_producer, event_producer, db_create_multiple_hosts, api_delete_host, mocker, db_get_hosts
+):
+    event_producer._kafka_producer.send.side_effect = (mocker.Mock(), mocker.Mock(**{"get.side_effect": KafkaError()}))
+
+    hosts = db_create_multiple_hosts(how_many=3)
+    host_id_list = [str(host.id) for host in hosts]
+
+    response_status, response_data = api_delete_host(",".join(host_id_list))
+
+    assert_response_status(response_status, expected_status=500)
```

Again, here. I think that letting the Kafka exception to bubble up and kill Flask is not the best thing we can do. If the error is caught and suggested, we can gracefullty return a _503 Service Unavailable_ response.

---

That's no different than how we handle e.g. sqlalchemy errors - we let them bubble up and "kill Flask". We can start catching those errors but that's outside the scope of this PR.